drivers/pir/pir.c

```diff
@@ -48,57 +50,68 @@ int pir_init(pir_t *dev, const pir_params_t *params)
     dev->active = false;
     dev->accum_active_time = 0;
     dev->start_active_time = 0;
-    dev->last_read_time = xtimer_usec_from_ticks64(xtimer_now64());
+    dev->last_read_time = xtimer_now_usec64();
 
-    if (gpio_init_int(dev->p.gpio, GPIO_IN_PD, GPIO_BOTH, pir_callback, dev)) {
-        return -1;
+    gpio_mode_t gpio_mode;
+    if (dev->p.active_high) {
+        gpio_mode = GPIO_IN_PD;
     }
-    return 0;
+    else {
+        gpio_mode = GPIO_IN_PU;
+    }
+
+    if (gpio_init_int(dev->p.gpio, gpio_mode, GPIO_BOTH, pir_callback, dev)) {
+        return PIR_NOGPIO;
+    }
+    return PIR_OK;
 }
 
 pir_event_t pir_get_status(const pir_t *dev)
 {
-    return ((gpio_read(dev->p.gpio) == 0) ? PIR_STATUS_LO : PIR_STATUS_HI);
+    return ((gpio_read(dev->p.gpio) == dev->p.active_high) ?
```

Comment:
careful: the gpio API returns either 0 or >0. You could change `gpio_read(..)` to `(gpio_read(..) > 0)`

Reply:
done